src/Models/Characters/Pawn.js

```diff
@@ -59,7 +57,21 @@ export default class Pawn{
         this.Position = targetPos;
     }
 
-    getTargets(opponentPawns){
+    getTargets(boardGrid){
+        let targetTiles = [];
+        
+        for(let dir of Object.values(Direction)){
+            targetTiles = targetTiles.concat(this.getAvailableAttacks(dir));
+        }
+        let targets = [];
+        targetTiles = targetTiles.filter(pos => pos.x >= 0 && pos.y >=0 && pos.x <12 && pos.y <12);
+        for(let tile of targetTiles){
+            let pawn = boardGrid[tile.x][tile.y].Occupants;
+            if(pawn && !targets.includes(pawn) && pawn.Occupants.Owner != this.Owner){
+                targets.push(pawn);
+            }
+        }
```

Comment:
Shouldn't the property be `Occupant` without the "s"?
`console.log(pawn)` returns undefined values so the loop never executes.
Is the last condition checking if the pawn doesn't belong to you? Doesn't it evaluate as
 `(boardGrid[tile.x][tile.y].Occupants).Occupants.Owner != this.Owner` instead of `pawn.Owner != this.Owner`?

Reply:
You're right! I'll fix that right now